### PR TITLE
[FEATURE]: purge_graph - purge whole graph excluding root

### PR DESCRIPTION
**Pull request by amadolid**

### CODE REVIEW

File: jac/jaclang/plugin/default.py [hunk 1]
```diff
@@ -561,6 +562,29 @@ def get_context() -> ExecutionContext:
         """Get current execution context."""
         return ExecutionContext.get()
 
+    @staticmethod
+    @hookimpl
+    def purge_graph(root: Optional[Root] = None) -> int:
```
Comment (ypkang) [inline jac/jaclang/plugin/default.py]: I'd just call this `reset_graph`, `purge` sounds too intense :)

Comment (amadolid) [reply jac/jaclang/plugin/default.py]: agreed haha

**Review by ChrisIsKing — APPROVED** [review]

File: jac-cloud/jac_cloud/core/architype.py [hunk 2]
```diff
@@ -387,15 +387,15 @@ def _pull(self) -> dict:
     def add_to_set(self, field: str, anchor: Anchor, remove: bool = False) -> None:
         """Add to set."""
         if field not in (add_to_set := self._add_to_set):
-            add_to_set[field] = {"$each": set()}
+            add_to_set[field] = {"$each": []}
 
-        ops: set = add_to_set[field]["$each"]
+        ops: list = add_to_set[field]["$each"]
```
Comment (ypkang) [inline jac-cloud/jac_cloud/core/architype.py:392]: What's the thinking for switching this from set to a list?

Comment (amadolid) [reply jac-cloud/jac_cloud/core/architype.py:392]: this is to retain the sequence of added objects

**Comment by amadolid** [comment]
REBASED

**Review by ypkang — APPROVED** [review]
@amadolid Can you resolve the conflict on this?

**Review by ypkang — APPROVED** [review]
LGTM

**Pull request merged**